src/main/java/com/ivoyant/AssessmentBook/Service/BookService.java

```diff
@@ -0,0 +1,38 @@
+package com.ivoyant.AssessmentBook.Service;
+
+import com.ivoyant.AssessmentBook.Model.Book;
+import com.ivoyant.AssessmentBook.Repository.BookRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class BookService {
+
+    @Autowired
+    private  BookRepository bookRepository;
+
+    public  Book upsert(Book book){
+        return bookRepository.save(book);
```

Comment:
Where are you handling all the exceptions? 

Is it in service layer or controller layer?

Reply:
handling the exceptions in service layer